PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-resource-connector-lib/src/main/java/com/microsoft/azure/toolkit/intellij/connector/ConnectorDialog.java

```diff
@@ -101,18 +104,25 @@ protected void onResourceOrConsumerTypeChanged(ItemEvent e) {
     protected void saveConnection(Connection<R, C> connection) {
         AzureTaskManager.getInstance().runLater(() -> {
             this.close(0);
-            final R resource = connection.getResource();
-            final C consumer = connection.getConsumer();
+            final R connectionResource = connection.getResource();
+            final C connectionConsumer = connection.getConsumer();
             final ConnectionManager connectionManager = this.project.getService(ConnectionManager.class);
             final ResourceManager resourceManager = ServiceManager.getService(ResourceManager.class);
-            final ConnectionDefinition<R, C> definition = ConnectionManager.getDefinitionOrDefault(resource.getType(), consumer.getType());
+            final ConnectionDefinition<R, C> definition = ConnectionManager.getDefinitionOrDefault(connectionResource.getType(), connectionConsumer.getType());
             if (definition.validate(connection, this.project)) {
-                resourceManager.addResource(resource);
-                resourceManager.addResource(consumer);
+                resourceManager.addResource(connectionResource);
+                resourceManager.addResource(connectionConsumer);
                 connectionManager.addConnection(connection);
                 final String message = String.format("The connection between %s and %s has been successfully created.",
-                        resource.toString(), consumer.toString());
+                        connectionResource.toString(), connectionConsumer.toString());
                 AzureMessager.getMessager().success(message);
+                // send connect success event.
+                ToolWindow toolWindow = ToolWindowManager.getInstance(project).getToolWindow("Azure Resource Connector");
+                if (!toolWindow.isVisible()) {
+                    toolWindow.show();
+                }
```

Comment:
it's some weird that ResourceConnector depends on `ToolWindow`.
they can decoupled with the event topic, isn't it?
I mean:  the toolwindow should subscribe the `resource-connection-added` topic, and then show itself.

Reply:
updated.